### PR TITLE
removed usage of setlogoption chain element

### PR DESCRIPTION
**Pull request by Mixaster995**
Signed-off-by: Mikhail Avramenko <avramenkomihail15@gmail.com>

## Description
Removed excessive chain element `setlopoption`

## Issue link
closes https://github.com/networkservicemesh/sdk/issues/1147

## How Has This Been Tested?

- [ ] Added unit testing to cover
- [x] Tested manually
- [ ] Tested by integration testing
- [ ] Have not tested

## Types of changes

- [ ] Bug fix
- [ ] New functionallity
- [ ] Documentation
- [x] Refactoring
- [ ] CI

### CODE REVIEW

File: pkg/registry/core/trace/context.go [hunk 1]
```diff
@@ -27,6 +27,10 @@ import (
 	"github.com/networkservicemesh/sdk/pkg/tools/log/spanlogger"
 )
 
+const (
+	loggedType = "registry"
```
Comment (denis-tingaikin) [inline pkg/registry/core/trace/context.go:31]: Let's use this inline
```suggestion
```

File: pkg/tools/log/logruslogger/logruslogger.go [hunk 2]
```diff
@@ -185,7 +185,7 @@ func (s *logrusLogger) WithField(key, value interface{}) log.Logger {
 
 // New - creates a logruslogger and returns it
 func New(ctx context.Context) log.Logger {
-	entry := logrus.WithFields(log.Fields(ctx))
+	entry := logrus.NewEntry(logrus.StandardLogger())
```
Comment (denis-tingaikin) [inline pkg/tools/log/logruslogger/logruslogger.go:188]: Why is this changed?

Also, who is using logruslogger.New?

**Review by denis-tingaikin — COMMENTED** [review]
Looks good. Added a few comments.

File: pkg/tools/log/logruslogger/logruslogger.go (hunk 2 above)
Comment (Mixaster995) [reply pkg/tools/log/logruslogger/logruslogger.go:188]: i removed `log.Fields(ctx)` method, so it has to be changed
`logruslogger.New` used in `cmd-...`s

Comment (denis-tingaikin) [reply pkg/tools/log/logruslogger/logruslogger.go:188]: Please note, cmd- previously could put fields into ctx. So please add logrus.Field variadic.

File: pkg/tools/log/logruslogger/logruslogger.go [hunk 3]
```diff
@@ -185,7 +185,7 @@ func (s *logrusLogger) WithField(key, value interface{}) log.Logger {
 
 // New - creates a logruslogger and returns it
 func New(ctx context.Context) log.Logger {
```
Comment (denis-tingaikin) [inline pkg/tools/log/logruslogger/logruslogger.go:187]:
```suggestion
func New(ctx context.Context, fields ... logrus.Field) log.Logger {
```

Comment (Mixaster995) [reply pkg/tools/log/logruslogger/logruslogger.go]: ok, done

File: pkg/tools/log/logruslogger/logruslogger.go [hunk 4]
```diff
@@ -183,9 +183,25 @@ func (s *logrusLogger) WithField(key, value interface{}) log.Logger {
...
+
+func toMap(fields ...Field) map[string]interface{} {
+	rv := make(map[string]interface{})
+
+	for _, f := range fields {
+		rv[f.key] = f.val
+	}
+
+	return rv
+}
```
Comment (denis-tingaikin) [inline pkg/tools/log/logruslogger/logruslogger.go:200]:
```suggestion
```

File: pkg/tools/log/logruslogger/logruslogger.go [hunk 5]
```diff
@@ -183,9 +183,25 @@ func (s *logrusLogger) WithField(key, value interface{}) log.Logger {
...
+
 // New - creates a logruslogger and returns it
-func New(ctx context.Context) log.Logger {
-	entry := logrus.NewEntry(logrus.StandardLogger())
+func New(ctx context.Context, fields ...Field) log.Logger {
```
Comment (denis-tingaikin) [inline pkg/tools/log/logruslogger/logruslogger.go:203]:
```suggestion
func New(ctx context.Context, fields ...logrus.Fields) log.Logger {
```

File: pkg/tools/log/logruslogger/logruslogger.go [hunk 6]
```diff
@@ -183,9 +183,25 @@ func (s *logrusLogger) WithField(key, value interface{}) log.Logger {
...
+	entry := logrus.WithFields(toMap(fields...))
```
Comment (denis-tingaikin) [inline pkg/tools/log/logruslogger/logruslogger.go:204]:
```suggestion
	var entry = logrus.NewEntry(logrus.New())
	
	for _, fieldMap := range fields {
		entry = entry.WithFields(fieldMap)
	}
```

**Review by denis-tingaikin — CHANGES_REQUESTED** [review]

File: pkg/tools/log/logruslogger/logruslogger.go (hunk 6 above)
Comment (Mixaster995) [reply pkg/tools/log/logruslogger/logruslogger.go]: ok, done

File: pkg/tools/log/logruslogger/logruslogger.go (hunk 5 above)
Comment (Mixaster995) [reply pkg/tools/log/logruslogger/logruslogger.go]: ok, done

File: pkg/tools/log/logruslogger/logruslogger.go (hunk 4 above)
Comment (Mixaster995) [reply pkg/tools/log/logruslogger/logruslogger.go]: ok, done

**Review by denis-tingaikin — APPROVED** [review]
